pkg/controllers/nodeclaim/lifecycle/launch_test.go

```diff
@@ -34,24 +35,43 @@ var _ = Describe("Launch", func() {
 	BeforeEach(func() {
 		nodePool = test.NodePool()
 	})
-	It("should launch an instance when a new NodeClaim is created", func() {
-		nodeClaim := test.NodeClaim(v1.NodeClaim{
-			ObjectMeta: metav1.ObjectMeta{
-				Labels: map[string]string{
-					v1.NodePoolLabelKey: nodePool.Name,
+	DescribeTable(
+		"Launch",
+		func(isNodeClaimManaged bool) {
+			nodeClaimOpts := []v1.NodeClaim{{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						v1.NodePoolLabelKey: nodePool.Name,
+					},
 				},
-			},
-		})
-		ExpectApplied(ctx, env.Client, nodePool, nodeClaim)
-		ExpectObjectReconciled(ctx, env.Client, nodeClaimController, nodeClaim)
+			}}
+			if !isNodeClaimManaged {
+				nodeClaimOpts = append(nodeClaimOpts, v1.NodeClaim{
+					Spec: v1.NodeClaimSpec{
+						NodeClassRef: &v1.NodeClassReference{
+							Group: "karpenter.k8s.aws",
+							Kind:  "EC2NodeClass",
+							Name:  "default",
+						},
+					},
+				})
+			}
+			nodeClaim := test.NodeClaim(nodeClaimOpts...)
+			ExpectApplied(ctx, env.Client, nodePool, nodeClaim)
+			ExpectObjectReconciled(ctx, env.Client, nodeClaimController, nodeClaim)
 
-		nodeClaim = ExpectExists(ctx, env.Client, nodeClaim)
+			nodeClaim = ExpectExists(ctx, env.Client, nodeClaim)
 
-		Expect(cloudProvider.CreateCalls).To(HaveLen(1))
-		Expect(cloudProvider.CreatedNodeClaims).To(HaveLen(1))
-		_, err := cloudProvider.Get(ctx, nodeClaim.Status.ProviderID)
-		Expect(err).ToNot(HaveOccurred())
-	})
+			Expect(cloudProvider.CreateCalls).To(HaveLen(lo.Ternary(isNodeClaimManaged, 1, 0)))
+			Expect(cloudProvider.CreatedNodeClaims).To(HaveLen(lo.Ternary(isNodeClaimManaged, 1, 0)))
+			if isNodeClaimManaged {
+				_, err := cloudProvider.Get(ctx, nodeClaim.Status.ProviderID)
+				Expect(err).ToNot(HaveOccurred())
+			}
+		},
+		Entry("should launch an instance when a new NodeClaim is created", true),
```

Comment:
I go either way on whether it's better to just make these separate It blocks. Personally, I would opt for different It() blocks rather than trying to collapse it into Tables

Reply:
The reason I went with `DescribeTables` is to ensure that we don't have test skew over time. If we add additional conditions to reach a state, it would be easy to miss adding those additional conditions to a negative test. By using the same test block, you ensure that the **only** reason the conditions wouldn't be set is that the node is unmanaged.